wdl-ast/src/v1/decls.rs

```diff
@@ -545,6 +656,46 @@ impl fmt::Display for Type {
     }
 }
 
+impl PartialOrd for Type {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Type {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        compare_types(self, other)
+    }
+}
+
+/// Compare all variants of Type.
+fn compare_types(a: &Type, b: &Type) -> std::cmp::Ordering {
```

Comment:
This is really hard to follow. Can it be rewritten using Rust `match` syntax?

Reply:
Excellent suggestion! I learned something new about `match` and it improves the readability here.